### DISCUSSION TITLE
Deepgram with Next JS

### DISCUSSION THREAD

**Discussion by Lipcsyy** [deepgram/community#456] [General help] [answered]
Hey!

I want to use Deepgram's API with Next JS. I've tried the example code:

const audioUrl = "https://static.deepgram.com/examples/Bueller-Life-moves-pretty-fast.wav";

// Initializes the Deepgram SDK
const deepgram = new Deepgram(deepgramApiKey);

console.log('Requesting transcript...')
console.log('Your file may take up to a couple minutes to process.')
console.log('While you wait, did you know that Deepgram accepts over 40 audio file formats? Even MP4s.')
console.log('To learn more about customizing your transcripts check out developers.deepgram.com.')

```
deepgram.transcription.preRecorded(
  { url: audioUrl },
  {smart_format: true, model: 'nova-2', language: 'en-US' },
)
.then((transcription) => {
  console.dir(transcription, {depth: null});
})
.catch((err) => {
  console.log(err);
});
```

But it seems to only work with the URL given. I saw that Deepgram can transcribe from youtube videos too. How can I change to code to work with youtube videos?

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by lukeocodes** [comment]
Hi @Lipcsyy, we actually have a YouTube demo using Next.js. I hope you find this helpful. It places the Deepgram integration on a serverside route file, and calls that from the client components. It isn't my best work, but... it works!

https://github.com/deepgram-devs/youtube-demo

**Reply by Lipcsyy** [reply]
Hey @lukeocodes !
Thank you for your answer. I've cloned your project, ran it and it didn't work for me. I tried implementing it on my side but I didn't understand what are you using SupaBase for. 

Thank you in advance.

**Reply by lukeocodes** [reply, marked as answer]
Supabase stores the results, you'll need to set up the environment for your own supabase credentials.

The main thing for you to see is this file, which is a serverside route in Next 13. You can omit the supabase code and resolve the request with the data from Deepgram

https://github.com/deepgram-devs/youtube-demo/blob/main/src/app/transcribe/api/route.ts

**Reply by Lipcsyy** [reply]
Thank you very much! Unfortunately, I have an issue with the metadata it says my transaction key is deprecated. Do you know why this occurs?

**Reply by lukeocodes** [reply]
Only that field is deprecated. I appreciate that it appears as though there is a problem, but you can ignore it completely.
